framework/include/problems/EigenProblem.h

```diff
@@ -37,17 +37,17 @@ class EigenProblem : public FEProblemBase
 
   virtual bool converged() override;
 
-  unsigned int getNEigenPairsRequired() { return _n_eigen_pairs_required; }
+  unsigned int getNEigenPairsRequired() const { return _n_eigen_pairs_required; }
   void setNEigenPairsRequired(unsigned int n_eigen_pairs)
   {
     _n_eigen_pairs_required = n_eigen_pairs;
   }
-  bool isGeneralizedEigenvalueProblem() { return _generalized_eigenvalue_problem; }
+  bool isGeneralizedEigenvalueProblem() const { return _generalized_eigenvalue_problem; }
   bool isNonlinearEigenvalueSolver() const;
   // silences warning in debug mode about the other computeJacobian signature being hidden
   using FEProblemBase::computeJacobian;
 
-  NonlinearEigenSystem & getNonlinearEigenSystem() { return *_nl_eigen; }
+  NonlinearEigenSystem & getNonlinearEigenSystem() const { return *_nl_eigen; }
```

Comment:
I don't like having this marked as `const` because it returns a non-const reference. This technically works because it marks the `_nl_eigen` pointer as `const` as opposed to the object that the pointer points to. But semantically speaking this method should be non-const